torch_xla/distributed/spmd/xla_sharding.py

```diff
@@ -16,24 +16,33 @@
 import numpy as np
 import functools
 import itertools
-from typing import Tuple, Union, List, Sequence, Any, Optional, Set
+from typing import Union, Sequence, Any, Optional
 from enum import IntEnum
 
 from torch.amp import custom_fwd, custom_bwd
 
+PartitionSpec = tuple[Union[tuple[Union[int, str], ...], int, str, None], ...]
+"""PartitionSpec describes the sharding of a tensor.
+
+Specifically, it is a tuple where each element references one or more device mesh
+axes, and instructs to shard the corresponding dimension of the input tensor over
+those mesh axes.
+"""
+
 
 class Mesh:
   """Describe the logical XLA device topology mesh and the underlying resources.
 
   Args:
-    device_ids (Union[np.ndarray, List]): A raveled list of devices (IDs) in a custom order. The list is reshaped
-        to an `mesh_shape` array, filling the elements using C-like index order.
+    device_ids: A raveled list of devices (IDs) in a custom order.
```

Comment:
Instead of "raveled" can we say "flattened"? I'm unfamiliar with the term, but I see there is a numpy function called `ravel`. It's best not to use a function as a verb, but rather use a verb that describes what the function is doing.

Reply:
yes. i didn't write this originally. and i was also confused by this word